### PR TITLE
VA-90347: Add Stats for FSR backend transform

### PR DESCRIPTION
**Pull request by maxx1128**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- *(Summarize the changes that have been made to the platform)*
  - This adds Datadog metrics to the new FSR backend transform that has been added, but is currently hidden behind a feature flag. It's being added to the same section where we have the respective logging for it.
- *(If bug, how to reproduce)* N/A
- *(What is the solution, why is this the solution?)* N/A
- *(Which team do you work for, does your team own the maintenance of this component?)*
  - You can find me on #debt-resolution on the Octo Slack
- *(If introducing a flipper, what is the success criteria being targeted?)* N/A

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
  - https://github.com/department-of-veterans-affairs/va.gov-team/issues/90347
- *Link to previous change of the code/bug (if applicable)* N/A
- *Link to epic if not included in ticket* N/A

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
  - We are not set up to collect stats on the new backend transform
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*
    - As of this writing, the aim is to start rolling out this feature on around 08/12/2024.

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
  - This currently impacts no areas of the site

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by RachalCassity — DISMISSED** [review]

**Comment by maxx1128** [comment]
@RachalCassity The failing spec related to the new tests is resolved now, so it should be good for another quick review.

**Review by RachalCassity — APPROVED** [review]

**Review by RachalCassity — CHANGES_REQUESTED** [review]
Fix failing specs please

**Comment by maxx1128** [comment]
I've confirmed that the failed tests were passing for me locally, so this should be good for re-review.

Screen Shot 2024-08-14 at 12 25 15 PM

**Review by RachalCassity — DISMISSED** [review]

File: modules/debts_api/app/controllers/debts_api/v0/financial_status_reports_controller.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@ class FinancialStatusReportsController < ApplicationController
       service_tag 'financial-report'
       before_action { authorize :debt, :access? }
 
+      STATSD_KEY_PREFIX = 'api.dmc'
```
Comment (digitaldrk) [inline modules/debts_api/app/controllers/debts_api/v0/financial_status_reports_controller.rb:13]: Only issue I see is here with the STATSD_KEY_PREFIX.

I see an example from the 1010EZ here:
app/controllers/v0/form1010_ezrs_controller.rb:24

Maybe we could do something like that and call it something like: `api.5655`

**Review by digitaldrk — APPROVED** [review]

**Pull request merged**